release-notes/4.0.0.md

```diff
@@ -0,0 +1,55 @@
+# Cucumber-ruby 4.0.0 release notes
+
+This is the first stable Cucumber release of the 4.x series. It's the first stable release for 2 years and there are pretty big changes since 3.1.2. Let's have a look at the biggest changes introduced by this release.
+
+## Added
+
+### new Gherkin syntax: Rules and Examples
+
+One of the most visible change in the new release is the support of the Gherkin 6+ syntax. This is still backward compatible with your existing features of course, but here are the changes introduced:
+ - the `Rule` keyword acts as a grouping of scenarios inside a feature
+ - the `Scenario Outline` keyword is not needed anymore when using `Examples` with a `Scenario``
+ - the `Scenario` keyword now has synonym: `Example`
+
+If you are familiar with [example mapping](https://cucumber.io/docs/bdd/example-mapping/), you will see how easy it is to translate your cards from an example mapping session into a `feature` file.
+
+### support of the cucumber-messages protocol
+
+[`cucumber-messages`](https://github.com/cucumber/cucumber/tree/master/messages#cucumber-messages) is the new format that is used by Cucumber to represent the various information emitted during an execution.
+
+This library satisfies multiple interests:
+ - same output for all `cucumber` and related implementation (cucumber-ruby, cucumber-jvm, cucumber-jss, SpecFlow ...)
+ - it relies on [`protobuf`](https://developers.google.com/protocol-buffers/), so it is fairly easy to generate a consumer in any language supported by `protobuf`.
+  - we have published libraries to consume this format for [`ruby`](https://rubygems.org/gems/cucumber-messages), [`Java`](https://mvnrepository.com/artifact/io.cucumber/messages), [`JavaScript](https://www.npmjs.com/package/@cucumber/messages) and [`C#`](https://www.nuget.org/packages/Cucumber.Messages/)
```

Comment:
Missing backtick for JavaScript - not sure why there are backticks in the first place.